ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -99,6 +99,12 @@ public abstract void configureInputJobProperties(TableDesc tableDesc,
     Map<String, String> jobProperties);
 
   /**
+   * This method is called to allw the StorageHandlers the chance to
+   * populate secret keys into the job's credentials.
+   */
+  public abstract void configureInputJobCredentials(TableDesc tableDesc, Map<String, String> secrets);
```

Comment:
Actually we handle the missing method at runtime just fine. This is just if you want to compile against latest version of hive. I think we can live with this. I'll check the imcompat flag.